models/staging/base/base_ga4__events.sql

```diff
@@ -1,61 +1,49 @@
-{% if var('static_incremental_days', false ) %}
-    {% set partitions_to_replace = ['current_date'] %}
-    {% for i in range(var('static_incremental_days')) %}
-        {% set partitions_to_replace = partitions_to_replace.append('date_sub(current_date, interval ' + (i+1)|string + ' day)') %}
-    {% endfor %}
-    {{
-        config(
-            pre_hook="{{ ga4.combine_property_data() }}" if var('property_ids', false) else "",
-            materialized = 'incremental',
-            incremental_strategy = 'insert_overwrite',
-            partition_by={
-                "field": "event_date_dt",
-                "data_type": "date",
-            },
-            partitions = partitions_to_replace,
-            cluster_by=['event_name']
-        )
-    }}
-{% else %}
-    {{
-        config(
-            materialized = 'incremental',
-            incremental_strategy = 'insert_overwrite',
-            partition_by={
-                "field": "event_date_dt",
-                "data_type": "date",
-            },
-            cluster_by=['event_name']
-        )
-    }}
-{% endif %}
---BigQuery does not cache wildcard queries that scan across sharded tables which means it's best to materialize the raw event data as a partitioned table so that future queries benefit from caching
-with source as (
+{% set partitions_to_replace = ['current_date'] %}
+{% for i in range(var('static_incremental_days')) %}
+    {% set partitions_to_replace = partitions_to_replace.append('date_sub(current_date, interval ' + (i+1)|string + ' day)') %}
+{% endfor %}
+{{
+    config(
+        pre_hook="{{ ga4.combine_property_data() }}" if var('property_ids', false) else "",
+        materialized = 'incremental',
+        incremental_strategy = 'insert_overwrite',
+        partition_by={
+            "field": "event_date_dt",
+            "data_type": "date",
+        },
+        partitions = partitions_to_replace,
+        cluster_by=['event_name']
+    )
+}}
+
+with source_daily as (
     select 
         {{ ga4.base_select_source() }}
-    {%  if var('frequency', 'daily') == 'streaming' %}
-        from {{ source('ga4', 'events_intraday') }}
-        where cast( _table_suffix as int64) >= {{var('start_date')}}
-    {% else %}
         from {{ source('ga4', 'events') }}
         where _table_suffix not like '%intraday%'
         and cast( _table_suffix as int64) >= {{var('start_date')}}
+    {% if is_incremental() %}
+        and parse_date('%Y%m%d', left(_TABLE_SUFFIX, 8)) in ({{ partitions_to_replace | join(',') }})
     {% endif %}
+),
```

Comment:
I think we need to add something like this below code to prevent data double-counting in the period between when GA creates a new events partition and before it deletes the equivalent events_intraday partition. @willbryant reported some issues related to this.

```
max_source_daily as (
  select max(event_date) as max_date
from source_daily
```
and then further down in the source_intraday block

```
where cast( _table_suffix as int64) >= {{var('start_date')}}
and cast(_table_suffix as int64) != max_source_daily.max_date
```
I think it has to be a does not equal condition here as this PR is supposed to also solve old events_intraday partitions that should be processed when the batch processing does not happen due to exceeding the 1 million event limit. So, we want to process old events_intraday partitions, just not the max partition when it equals the events max partition.

This assumes two things:
1. That the max events partition is always complete
2. Google will delete the intraday partition before the next day's processing.

I think 1 is a safe assumption.

For 2, it will still be possible for this to happen.

Day 1: partitions exist for `events_Day 1 - 1` and `events_intradayDay 1 - 1` and the package processes Day 1 - 1 as desired
Day 2: partitions still exist for  `events_Day 1 - 1` and `events_intradayDay 1 - 1` (equivalent of Day 2 -2) and Day 1 data gets doubled

I think this is unlikely, certainly much less likely than not suppressing the max partition from source_daily on source_intraday.

Reply:
@dgitis what's the harm  in letting the dedupe logic at the end of `base_ga4__events` catch this, if there are duplicate events across daily & intraday?

Reply:
The user ltv data comes in its own column that's not deduped so that's not an issue. 

I think it would be fine. It might cost a little extra processing, but we don't need everything to be perfect as long as it's functional.

We might run into a problem in the future if Google adds any processed data to `event_params`. 

Is the `google / cpc` traffic coming through as `google / organic` problem only a problem on intraday? Or is it there for batch as well?

If it's a problem on both then I think we can ignore my comment.